xaitk_saliency/utils/tools/drise_from_config.py

```diff
@@ -0,0 +1,130 @@
+import numpy as np
+from typing import Optional, Collection
+
+from smqtk_detection.interfaces.detect_image_objects import DetectImageObjects
+from smqtk_core.configuration import from_config_dict
+
+from xaitk_saliency.impls.perturb_image.rise import RISEGrid
+from xaitk_saliency.impls.gen_detector_prop_sal.drise_scoring import DRISEScoring
+from xaitk_saliency.utils.masking import occlude_image_batch
+from xaitk_saliency.utils.detection import format_detection
+
+
+def drise_from_config(
+    ref_imgs: Collection[np.ndarray],
+    config_dict: dict,
+    num_perturbations: int = 100,
+    verbose: Optional[bool] = False,
+) -> Collection[np.ndarray]:
+    """
+    Generate saliency maps for detections in each of the given images.
+    Detections are performed by and implementation of ``DetectImageObjects``
+    for which a configuration dictionary is passed. This implementation must be
+    discoverable by ``DetectImageObjects.get_impls()``.
+
+    The DRISE algorithm returns saliency maps for each detection in the image,
+    resulting in an output of shape [nDets, imgHeight, imgWidth] for each
+    respective image.
+
+    :param ref_imgs: List of images to generate saliency maps for.
+    :param config_dict: Configuration dictionary for ``DetectImageObjects``
+        implementation.
+    :param num_perturbations: Number of perturbed images to use for the
+        saliency map generation. Defaults to 100 if not provided.
+    :param verbose: Option to print out progress statements. Default is false.
+
+    :returns: List of saliency map matrices, each with shape
+        [nDets, imgHeight, imgWidth] respective to their corresponding image.
+    """
+
+    def dets_to_formatted_mat(dets):  # type: ignore
+        """
+        This helper function takes detections as returned by an implementation
+        of DetectImageObjects and returns a matrix formatted for use with the
+        DRISE algorithm, as returned by the format_detection() function.
+        """
+
+        if len(dets) == 0:
+            return np.asarray([])
+
+        bbox_mat = np.asarray([
+            [
+                bbox.min_vertex[0],
+                bbox.min_vertex[1],
+                bbox.max_vertex[0],
+                bbox.max_vertex[1]
+            ] for bbox in [det[0] for det in dets]
+        ])
+
+        class_mat = np.asarray([
+            list(class_dict.values())
+            for class_dict in [det[1] for det in dets]
+        ])
+
+        return format_detection(bbox_mat, class_mat)
+
+    # DetectImageObjects implementation
+    blackbox_detector = from_config_dict(config_dict, DetectImageObjects.get_impls())
+
+    img_perturber = RISEGrid(num_perturbations, 8, 0.5, seed=0, threads=4)
+    sal_generator = DRISEScoring()
+
+    # List of saliency maps for all images
+    sal_maps_set = []
```

Comment:
Call this a list. Calling it a set is a little misleading given that there is a literal set-type.